include/boost/detail/winapi/dll.hpp

```diff
@@ -18,83 +19,186 @@
+}}} // namespace boost::detail::winapi
 
 #if !defined( BOOST_NO_ANSI_APIS )
 BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 LoadLibraryA(boost::detail::winapi::LPCSTR_ lpFileName);
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
+LoadLibraryExA(
+    boost::detail::winapi::LPCSTR_ lpFileName,
+    boost::detail::winapi::HANDLE_ hFile,
+    boost::detail::winapi::DWORD_ dwFlags
+);
+
+BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 GetModuleHandleA(boost::detail::winapi::LPCSTR_ lpFileName);
+
+BOOST_SYMBOL_IMPORT boost::detail::winapi::DWORD_ WINAPI
+GetModuleFileNameA(
+    boost::detail::winapi::HMODULE_ hModule,
+    boost::detail::winapi::LPSTR_ lpFilename,
+    boost::detail::winapi::DWORD_ nSize
+);
 #endif
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 LoadLibraryW(boost::detail::winapi::LPCWSTR_ lpFileName);
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
+LoadLibraryExW(
+    boost::detail::winapi::LPCWSTR_ lpFileName,
+    boost::detail::winapi::HANDLE_ hFile,
+    boost::detail::winapi::DWORD_ dwFlags
+);
+
+BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 GetModuleHandleW(boost::detail::winapi::LPCWSTR_ lpFileName);
 
+BOOST_SYMBOL_IMPORT boost::detail::winapi::DWORD_ WINAPI
+GetModuleFileNameW(
+    boost::detail::winapi::HMODULE_ hModule,
+    boost::detail::winapi::LPWSTR_ lpFilename,
+    boost::detail::winapi::DWORD_ nSize
+);
+
+
 BOOST_SYMBOL_IMPORT boost::detail::winapi::BOOL_ WINAPI
 FreeLibrary(boost::detail::winapi::HMODULE_ hModule);
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::FARPROC_ WINAPI
 GetProcAddress(boost::detail::winapi::HMODULE_ hModule, boost::detail::winapi::LPCSTR_ lpProcName);
-}
-#endif
 
-namespace boost {
-namespace detail {
-namespace winapi {
+BOOST_SYMBOL_IMPORT boost::detail::winapi::FARPROC_ WINAPI
+VirtualQuery(
+    boost::detail::winapi::LPCVOID_ lpAddress,
+    boost::detail::winapi::MEMORY_BASIC_INFORMATION_* lpBuffer,
+    boost::detail::winapi::ULONG_PTR_ dwLength
+);
+
+} // extern "C"
+#endif // #if !defined( BOOST_USE_WINDOWS_H )
+
+namespace boost { namespace detail { namespace winapi {
 
 #if defined( BOOST_USE_WINDOWS_H )
 typedef ::FARPROC FARPROC_;
 typedef ::NEARPROC NEARPROC_;
 typedef ::PROC PROC_;
+typedef ::MEMORY_BASIC_INFORMATION MEMORY_BASIC_INFORMATION_;
+
+const DWORD_ DONT_RESOLVE_DLL_REFERENCES_           = DONT_RESOLVE_DLL_REFERENCES;
+const DWORD_ LOAD_IGNORE_CODE_AUTHZ_LEVEL_          = LOAD_IGNORE_CODE_AUTHZ_LEVEL;
+const DWORD_ LOAD_LIBRARY_AS_DATAFILE_              = LOAD_LIBRARY_AS_DATAFILE;
+const DWORD_ LOAD_LIBRARY_AS_DATAFILE_EXCLUSIVE_    = LOAD_LIBRARY_AS_DATAFILE_EXCLUSIVE;
+const DWORD_ LOAD_LIBRARY_AS_IMAGE_RESOURCE_        = LOAD_LIBRARY_AS_IMAGE_RESOURCE;
+const DWORD_ LOAD_WITH_ALTERED_SEARCH_PATH_         = LOAD_WITH_ALTERED_SEARCH_PATH;
 #endif // defined( BOOST_USE_WINDOWS_H )
 
+// We always define the following constants by explicit values, because
```

Comment:
If these constants are only supported from Windows 8, their definition should probably be conditioned on `BOOST_USE_WINAPI_VERSION`, shouldn't it?